File: apps/android/core/model/src/main/kotlin/app/gauja/core/model/servers/Preferences.kt
// SPDX-FileCopyrightText: 2026 Gauja contributors
// SPDX-License-Identifier: AGPL-3.0-or-later
package app.gauja.core.model.servers

enum class ThemeMode {
    DARK,
    LIGHT,
    SYSTEM,
}

data class Preferences(
    val theme: ThemeMode = ThemeMode.DARK,
    val activeProfileId: ProfileId? = null,
    val reduceMotion: Boolean = false,
)

data class DiscoverPreferences(val region: String? = null, val providerIds: Set<Int> = emptySet())

File: apps/ios/Packages/Model/Sources/Model/servers/Preferences.swift
// SPDX-FileCopyrightText: 2026 Gauja contributors
// SPDX-License-Identifier: AGPL-3.0-or-later
public enum ThemeMode: String, Sendable, Codable { case dark, light, system }

public struct Preferences: Sendable, Equatable, Codable {
    public var theme: ThemeMode = .dark
    public var activeProfileID: ProfileID?
    public var reduceMotion = false
    public init() {}
}

public struct DiscoverPreferences: Sendable, Equatable, Codable {
    public var region: String?
    public var providerIDs: Set<Int> = []
    public init() {}
}
